3rdparty/libcxx/CMakeLists.txt

```diff
@@ -48,9 +48,6 @@ maybe_build_using_clangw(libcxx)
 
 add_dependencies(libcxx libcxx_includes)
 
-set_target_properties(libcxx PROPERTIES
-  POSITION_INDEPENDENT_CODE ON)
```

Comment:
For my understanding, does this actually translate into `-fPIC`? The [CMake docs](https://cmake.org/cmake/help/latest/module/CheckPIESupported.html?highlight=pie#command:check_pie_supported) seem to suggest that it expects `POSITION_INDEPENDENT_CODE` to be used with a linker that supports PIE, which is weird. 

Assuming this should be removed, does an explicit `target_compile_options(libcxx PUBLIC -fPIE)` need to be added in replacement?

Reply:
`POSITION_INDEPENDENT_CODE` translates to -fPIC for libraries and -fPIE for executables (and is otherwise just kind of [buggy](https://gitlab.kitware.com/cmake/cmake/issues/14983)). We want -fPIE for all enclave code; so it is in the `INTERFACE` compile options of oecore. Since libcxx links to oecore, the flag is propagated and does not need to be manually added here. (This is true because we build libcxx directly in our CMake graph.)

Reply:
From @CodeMonkeyLeet:

> Thanks for the explanation!

> A couple of follow-up questions:

> Why does mbedtls need to specify -fPIE in its CMakeLists.txt then? It does link against oecore, but is that import broken because ExternalProject_Add doesn’t pick up it up from just a DEPENDS on oecore? Might be worth pointing out as a comment.

> Why does musl need to specify -fPIE in its CMakeLists.txt at all? Does that actually affect the header generation? It looks like most of the dependent builds will get it through oelibc (which gets it from oecore).

Reply:
@CodeMonkeyLeet You are correct, though I'm not sure how helpful a comment here will be. The mbedtls build is via ExternalProject_Add, meaning it is an _external_ build and not part of our CMake build, and it does _not_ link to oecore. It's just that `oecore` is in its `INTERFACE` as a dependency for other targets linking to `mbedtls`.

It probably doesn't affect header generation of musl, that's true. I'll remove it.